File: program/program/convert-binary-to-hexadecimal/convert_binary_to_hexadecimal.go
package main

import (
    "fmt"
    "strconv"
)

func parseBinToHex(s string) string {
    ui, err := strconv.ParseUint(s, 2, 64)
    if err != nil {
        return "error"
    }

    return fmt.Sprintf("%x", ui)
}

func main() {
  var value string
  fmt.Println("Enter binary value:")
  fmt.Scan(&value)
  fmt.Println(parseBinToHex(value))
}

File: program/program/convert-binary-to-hexadecimal/convert-binary-to-hexadecimal.py
# Input a binary number and remove any white spaces
binary_num = (input("Enter a binary number: ")).replace(" ", "")

try:
    # Convert to integer
    int_num = int(binary_num, 2)

    # Convert to hexadecimal and format correctly
    hex_num = format(int_num, "x").upper()

    # Print result
    print(hex_num)

except:
    # Error
    print("Please input a valid binary number.")
